org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/semantictokens/SemanticTokensLegend.java

```diff
@@ -13,22 +13,28 @@
  *******************************************************************************/
 package org.eclipse.jdt.ls.core.internal.semantictokens;
 
-import java.util.List;
+import java.util.Arrays;
 
 public class SemanticTokensLegend {
-	private final List<String> tokenTypes;
-	private final List<String> tokenModifiers;
+	public static final SemanticTokensLegend INSTANCE = new SemanticTokensLegend();
```

Comment:
Oh, I see the original purpose of SemanticTokenManager now. But I just thought it was a bit unnecessary after the changes I made, since it was only used to get the SemanticTokensLegend instance. I can re-implement the lazy-loading in SemanticTokensLegend instead.

But now that I think about it, is there any reason to keep the legend in-memory at all in Java-land? Couldn't we just create a new instance when requested by the client? Because the server never uses the legend, and the client only needs to request it once and then keep it in it's own memory. Does the server have any reason to keep holding a static instance of the legend then?

Reply:
>  is there any reason to keep the legend in-memory at all in Java-land?

It's a good finding. Indeed, we don't have to have the Legend, as it simply holds all possible values of `TokenType` and `TokenModifier`.

> Does the server have any reason to keep holding a static instance of the legend then?

When client talks to the server, `Legend` is their contract, and it was used to encode tokens. With consistency in mind, I just implemented in that way.